lstchain/irf/hdu_table.py

```diff
@@ -0,0 +1,293 @@
+import numpy as np
+import os
+import logging
+
+from lstchain.reco.utils import camera_to_altaz
+
+import astropy.units as u
+from astropy.table import Table, Column, vstack, QTable
+from astropy.io import fits
+from astropy.time import Time
+
+__all__ = [
+    'create_obs_hdu_index',
+    'create_event_list'
+    ]
+
+log = logging.getLogger(__name__)
+
+DEFAULT_HEADER = fits.Header()
+DEFAULT_HEADER["HDUDOC"] = "https://github.com/open-gamma-ray-astro/gamma-astro-data-formats"
+DEFAULT_HEADER["HDUVERS"] = "0.2"
+DEFAULT_HEADER["HDUCLASS"] = "GADF"
+
+#IRF should be separate?
+def create_obs_hdu_index(filename_list, fits_dir):
+    """
+    Create the obs table and hdu table (below some explanation)
+
+    A two-level index file scheme is used in the IACT community to allow an arbitrary folder structures
+    For each directory tree, two files should be present:
+    **obs-index.fits.gz**
+    (defined in http://gamma-astro-data-formats.readthedocs.io/en/latest/data_storage/obs_index/index.html)
+    **hdu-index.fits.gz**
+    (defined in http://gamma-astro-data-formats.readthedocs.io/en/latest/data_storage/hdu_index/index.html)
+
+    obs-index contains the single run informations e.g.
+    (OBS_ID, RA_PNT, DEC_PNT, ZEN_PNT, ALT_PNT)
+    while hdu-index contains the informations about the locations of the other HDU (header data units)
+    necessary to the analysis e.g. A_eff, E_disp and so on...
+        http://gamma-astro-data-formats.readthedocs.io/en/latest/
+
+    This function will create the necessary data format, starting from the path that contains the DL3
+    converted fits file.
+
+    Parameters
+    ----------
+    filename_list : list
+        list of filenames
+    fits_dir : str
+        directory containing the fits file
+    """
+
+    hdu_tables = []
+    obs_tables = []
+
+    #loop through the files
+    for file in filename_list:
+        filepath = (fits_dir/file)
+        if filepath.is_file():
+            try:
+                event_table = Table.read(filepath, hdu="EVENTS")
+                gti_table = Table.read(filepath, hdu="GTI")
+                pointing_table = Table.read(filepath, hdu="POINTING")
+            except Exception:
+                log.error(f"fits corrupted for file {file}")
+                continue
+        else:
+            log.error(f"fits {file} doesn't exist")
+            continue
+
+        #The column names for the table follows the scheme as shown in
+        #https://gamma-astro-data-formats.readthedocs.io/en/latest/general/HDU_CLASS.html
+        ###############################################
+        # Event list
+        t_events = Table(
+                {'OBS_ID':[event_table.meta['OBS_ID']],'HDU_TYPE':['events'],'HDU_CLASS':['events'],
+                    'HDU_CLASS2':[''],'HDU_CLASS3':[''],'HDU_CLASS4':[''],'FILE_DIR':[''],'FILE_NAME': [file],
```

Comment:
'events' should be upper case here and I don't think you need to add empty `HDU_CLASSX` header entries